tsweb/tsweb.go

```diff
@@ -396,6 +397,32 @@ func VarzHandler(w http.ResponseWriter, r *http.Request) {
 				dump(name+"_", kv)
 			})
 			return
+		case PrometheusMetricsReflectRooter:
+			root := v.PrometheusMetricsReflectRoot()
+			rv := reflect.ValueOf(root)
+			if rv.Type().Kind() == reflect.Ptr {
+				if rv.Pointer() == 0 {
+					return
+				}
+				rv = rv.Elem()
+			}
+			if rv.Type().Kind() != reflect.Struct {
+				fmt.Fprintf(w, "# skipping expvar %q; unknown root type\n", name)
+				return
+			}
+			foreachExportedStructField(rv, func(fieldOrJSONName, promTyp string, rv reflect.Value) {
+				mname := name + "_" + fieldOrJSONName
+				switch rv.Kind() {
+				case reflect.Int, reflect.Int64:
+					fmt.Fprintf(w, "# TYPE %s %s\n%s %v\n", mname, promTyp, mname, rv.Int())
+				case reflect.Float64:
+					fmt.Fprintf(w, "# TYPE %s %s\n%s %v\n", mname, promTyp, mname, rv.Float())
+				case reflect.Struct:
+					dump(name+"_", expvar.KeyValue{Key: fieldOrJSONName, Value: expVarPromStructRoot{rv.Addr().Interface()}})
```

Comment:
rv.Addr() only works if the parent struct is addressable. In the event that the parent struct is not a pointer, then this panics.

I believe that's possible since we permit both structs and pointer to structs at the top-level. I think the former case would trigger a panic here.

Reply:
Done.